controllers/capabilities/mapping_rules.go

```diff
@@ -120,28 +132,19 @@ func (t *ProductThreescaleReconciler) getExistingMappingRules() (map[string]thre
 		return nil, fmt.Errorf("Error getting product [%s] mappingrules: %w", t.resource.Spec.SystemName, err)
 	}
 	for _, item := range existingList.MappingRules {
-		key := fmt.Sprintf("%s:%s", item.Element.HTTPMethod, item.Element.Pattern)
+		key := fmt.Sprintf("%s:%s:%s", item.Element.HTTPMethod, item.Element.Pattern, strconv.FormatInt(item.Element.MetricID, 10))
```

Comment:
Going by position in general is better.
When ranging through mapping rules from CR, we always go top down (top being 0, so position+1)
When ranging through mapping rules from 3scale, we can use item.Element.Position to get position.